frontend/src/components/Input/Input.tsx

```diff
@@ -0,0 +1,56 @@
+import {
+  forwardRef,
+  Icon,
+  Input as ChakraInput,
+  InputGroup,
+  InputProps as ChakraInputProps,
+  InputRightElement,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { omit } from '@chakra-ui/utils'
+
+import { BxsCheckCircle } from '~assets/icons/BxsCheckCircle'
+
+export interface InputProps extends ChakraInputProps {
+  /**
+   * Whether the input is in a prefilled state.
+   */
+  isPrefilled?: boolean
+  /**
+   * Whether the input is in a success state.
+   */
+  isSuccess?: boolean
+}
+
+export const Input = forwardRef<InputProps, 'input'>((props, ref) => {
+  const inputStyles = useMultiStyleConfig('Input', props)
+
+  // Omit extra props so they will not be passed into the DOM and trigger
+  // React warnings.
+  const inputProps = omit(props, ['isSuccess', 'isPrefilled'])
+
+  // Return normal input component if not success state.
+  if (!props.isSuccess) {
+    return <ChakraInput ref={ref} {...inputProps} sx={inputStyles.field} />
+  }
+
+  return (
+    // InputGroup is required for InputRightElement to retrieve the correct
+    // style props. Will crash if not included.
+    <InputGroup>
+      <ChakraInput ref={ref} {...inputProps} sx={inputStyles.field} />
+      <InputRightElement sx={inputStyles.success}>
+        <Icon as={BxsCheckCircle} />
+      </InputRightElement>
+    </InputGroup>
```

Comment:
Could i clarify on why we chose to do this? Testing on chakra's website seems to suggest that there are styling conflicts and consumers of our component might mistakenly think that this is a base `Input` in success state and compose it as part of an `InputGroup` only to discover that it's actually not.